integration_testing/features/super-admin/super-admin-import-new-content-channel-from-studio.feature

```diff
@@ -9,46 +11,81 @@ Feature: Super admin imports content from Studio
       And I see the total number and size of <channel> channel resources
       And I see the value for *Drive space available*     
       And I see 0 resources from <channel> channel are listed as *On your device*
+      And I see the list of topics for the <channel> channel
 
-  Scenario: When I check the *Select all* checkbox
-    Then I see the checkboxes for all the topics are checked
+  Scenario: Navigate the topic tree
+    When I click the <topic> topic link
+    Then I see the list of subtopics for the <topic> topic
+      And I see the channel name as a breadcrumb link
+      And I see the topic name as a breadcrumb next to the channel name
+    When I click the <subtopic> subtopic link
+    Then I see the list of subtopics for the <subtopic> subtopic
+      And I see the subtopic name as a link
+      And I see the subtopic name as a breadcrumb next to the topic name
+    When I click the topic name link in the breadcrumb
+    Then I see the topic tree page
+    When I click the channel name link
+    Then I see the *Select content from '<channel>'* page
+
+  Scenario: Select all topics or subtopics
+    When I check the *Select all* checkbox
+    Then I see the checkboxes for all the topics or subtopics are checked
       And I see the *Import* button is active
-      And I see the values for *Content selected* increase
+      And I see the number of *resources selected* flag for each topic checkbox
+      And I see the sum of size and number of resources selected in the *Content selected*
+
+  Scenario: Unselect all topics or subtopics
     When I uncheck the *Select all* checkbox
+    Then I see the checkboxes for all the topics or subtopics are unchecked
+      And I do not see the number of *resources selected* flag for unchecked topics
+    Given there are other topics checked in the topic tree
+    Then I see the *Import* button is still active
+      And I see the values for *Content selected* is deducted with the values from the unchecked topics
+    Given that no other topics are checked in the topic tree
     Then I see the *Import* button is inactive
       And I see the values for *Content selected* is 0
 
-  Scenario: When I check the <topic> topic checkbox
+  Scenario: Check a topic or subtopic
+    When I check a <topic> topic checkbox
     Then I see the *Import* button is active
+      And I see the number of *resources selected* flag for the selected topic checkbox
       And I see the values for *Content selected* increase
-    When I uncheck the <topic> topic checkbox
+
+  Scenario: Uncheck a topic or subtopic
+    When I uncheck a topic checkbox
+    Then I do not see the number of *resources selected* flag for the unchecked topic
+    Given there is another topic checked
+    Then I see the *Import* button is still active
+      And I see the values for *Content selected* is deducted with the values from the unchecked topic
+    Given it is the only topic checked
```

Comment:
I know we are not following the "standard" Gherkin syntax by the letter, but not sure if it's kosher to have `Givens` pop-up in the middle of a scenario... 🤔 
Maybe put them in the comment after/above the step?

Reply:
Thanks for the review! Will edit as per your comments. 😸